code/game/objects/items/weebstick.dm

```diff
@@ -0,0 +1,194 @@
+	//There could be a way to go around that through finding out the turf the belt is on, but a owner link is easier & fits the theme, too.
+	var/dash_sound = 'sound/weapons/unsheathed_blade.ogg'
+	var/beam_effect = "blood_beam"
+	var/phasein = /obj/effect/temp_visual/dir_setting/cult/phase
+	var/phaseout = /obj/effect/temp_visual/dir_setting/cult/phase
+
+/obj/item/weebstick/Initialize()
+	. = ..()
+	AddComponent(/datum/component/butchering, 25, 90, 5) //Not made for scalping victims, but will work nonetheless
+
+/obj/item/weebstick/suicide_act(mob/user)
+	if(prob(50))
+		user.visible_message("<span class='suicide'>[user] carves deep into [user.p_their()] torso! It looks like [user.p_theyre()] trying to commit seppuku...</span>")
+	else
+		user.visible_message("<span class='suicide'>[user] carves a grid into [user.p_their()] chest! It looks like [user.p_theyre()] trying to commit sudoku...</span>")
+	return (BRUTELOSS)
+
+/obj/item/weebstick/attack(mob/living/victim)
+	if(iscarbon(victim))
+		var/prevstat = victim.stat //save original state
+		. = ..()
+		if(victim.mind && victim.stat == SOFT_CRIT && victim.stat != prevstat) //when they're in crit but they weren't before the attack, prime it
+			primed = TRUE
+	else
+		var/prevstat = victim.stat
+		. = ..()
+		if(victim.mind && victim.stat == DEAD && victim.stat != prevstat) //sadly silicons/simplemobs have to die for the weapon to be primed.
+			primed = TRUE
+
+/obj/item/weebstick/attack_self(mob/living/carbon/user)
+	. = ..()
+	if (user != owner)
+		owner = user
+		var/opposite_active_arm = BODY_ZONE_R_ARM
+		if(!owner.active_hand_index % 2) // no direct function for getting the opposite body zone of the current active hand. Uses the hand index to be able to assign the correct zone;
+		// since you can have more than two hands, the modulo returns 0 or 1 depending if it's a right or left one.
+			opposite_active_arm = BODY_ZONE_L_ARM
+		owner.apply_damage(20, BRUTE, opposite_active_arm, FALSE, TRUE)
+		primed = TRUE //freebie
+	//Becoming the owner damages the user 20 damage and allows them to use the dash attack. anyone can become owner, but need sheathe for the dash.
+
+/obj/item/weebstick/on_exit_storage(datum/component/storage/concrete/S)
+	var/obj/item/storage/belt/weebstick/B = S.real_location()
+	if(istype(B))
+		playsound(B, 'sound/items/unsheath.ogg', 25, TRUE)
+		B.update_icon()
+
+/obj/item/weebstick/on_enter_storage(datum/component/storage/concrete/S)
+	var/obj/item/storage/belt/weebstick/weebbelt = S.real_location()
+	if(istype(weebbelt))
+		playsound(weebbelt, 'sound/items/sheath.ogg', 25, TRUE)
+		weebbelt.update_icon()
+		if(primed) // if found on owner & primed, then allow for dash. else, no. Missing owner check.
+			RegisterSignal(owner, COMSIG_MOB_CLICKON, .proc/predash)
+	return
+
+/obj/item/weebstick/proc/predash(mob/user, atom/location, params)
+	SIGNAL_HANDLER
+
+	if(!(location in view(user.client.view,user)))
```

Comment:
Don't assume client and you can just check mobs view

Reply:
ah, didn't know that - modified!